test/parallel/test-http-parser-reuse-domain.js

```diff
@@ -0,0 +1,80 @@
+'use strict';
+
+const common = require('../common');
+const domain = require('domain');
+const http = require('http');
+
+const agent = new http.Agent({
+  // keepAlive is important here so that the underlying socket of all requests
+  // is reused and tied to the same domain.
+  keepAlive: true
```

Comment:
Can we have an assertion that checks that the connection is the same for all requests?

Reply:
Good point!

Instead of an assertion, which can throw an exception and thus trigger a different code path with regards to domain error handling that would make this test really difficult to write, do you mind if, as for other failure cases in this test, we set the exit code to 1?

Reply:
@misterdjules Yes, that sounds good to me as well :)